### ISSUE TITLE
Env var to set app URL

### ISSUE THREAD

**Issue by m3nu** [stonith404/pingvin-share#258]
### 🙋‍♂️ Question

Hi, this is a very well-made app and I found it via Jeremy from [noted.lol](https://noted.lol/). Currently looking into adding it on  [pikapods.com](https://www.pikapods.com/) for 1-click deployment.

Currently I'm running into an issue with setting the app URL to use. I see an option for `APP_URL` in the database, but it's not an env var that's read in.

How would I best set the external app URL without this env var? Currently it defaults to `http://localhost:3000`. I know the URL can be changed via UI, but for most other apps, we can simply set it via env var.

Would you consider a pull request to update the `APP_URL` setting from an env var? Generally supporting all env vars from here would be great:

https://github.com/stonith404/pingvin-share/issues/22#issuecomment-1296183651

**Comment by stonith404** [comment]
Yeah I think it makes more sense to store this in an env var instead of changing it in the UI. In my opinion server specific configuration should be done with env variables and app specific configurations in the web UI.

What do you think?

**Comment by m3nu** [comment]
Totally agree of course. That was why I brought it up. 😀

From all the apps I've set up, most of them set the final domain via env var. Or at least offer this option if some kind of auto-detection doesn't work correctly.

**Comment by m3nu** [comment]
Circling back to this, since I'd love to add this app without messing with the database during each edit: Did you end up using the `APP_URL` env var anywhere? I only see it in SQL. If it's available, I'd also add it in the README here:

https://github.com/stonith404/pingvin-share/tree/main#environment-variables

**Comment by stonith404** [comment]
Oh yeah sorry, I don't know why I have closed this issue. I didn't implement the env var yet :/

**Comment by matbgn** [comment]
Thank you for removing the stale label, I confirm that it will be amazing to have this env var ;-)

**Comment by brunnels** [comment]
+1 for OIDC in env vars.  I'm deploying with flux in kubernetes so making everything configurable from environment vars or a config file would be ideal.

**Comment by stonith404** [comment]
It's now possible to configure every configuration in a yaml file. See the [docs entry](https://stonith404.github.io/pingvin-share/setup/configuration#yaml-file).

**Comment by TheDarkula** [comment]
@stonith404 Are the environment variables for OIDC/Oauth2 somewhere other than [here](https://stonith404.github.io/pingvin-share/setup/configuration/#environment-variables)? I cannot find any documentation regarding those.
